pkg/provisioner/controller.go

```diff
@@ -291,6 +293,103 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	// to control access to static buckets via RBAC rules on storage classes.
 	isDynamicProvisioning := isNewBucketByStorageClass(class)
 
+	// Re-Get the claim in order to shorten the race condition where the claim was deleted after provisioning started
+	obc, err = claimForKey(key, c.libClientset)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
Does this test imply that the finalzer just added above was not present? Otherwise the OBC could not actually be deleted. Are you seeing a window where we set the finalizer but it's not persisted and thus this GET is required?

Reply:
This code block I did not add but merely moved up from below. Maybe this block isn't necessary and I can remove it?

Reply:
I'm guessing the original location for this block made more sense. What is your reason for moving it? I am not sure it's even needed now but haven't checked more deeply to figure that out yet...

Reply:
Should I move this to before `setOBCMetaFields`? Remove it? I'll follow your recommendation on this.